protocol/x/clob/e2e/conditional_orders_test.go

```diff
@@ -958,6 +958,1125 @@ func TestConditionalOrder(t *testing.T) {
 	}
 }
 
+func TestConditionalOrder_TriggeringUsingMatchedPrice(t *testing.T) {
+	tests := map[string]struct {
+		subaccounts          []satypes.Subaccount
+		ordersForFirstBlock  []clobtypes.Order
+		ordersForSecondBlock []clobtypes.Order
+
+		expectedInTriggeredStateAfterBlock map[uint32]map[clobtypes.OrderId]bool
+
+		// these expectations are asserted after all blocks are processed
+		expectedExistInState    map[clobtypes.OrderId]bool
+		expectedOrderFillAmount map[clobtypes.OrderId]uint64
+		expectedSubaccounts     []satypes.Subaccount
+	}{
+		"TakeProfit/Buy conditional order is placed and not triggered by matched price": {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_100_000USD,
+				constants.Carl_Num1_100000USD,
+				constants.Dave_Num1_500000USD,
+			},
+			ordersForFirstBlock: []clobtypes.Order{
+				constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_TP_49995,
+			},
+			ordersForSecondBlock: []clobtypes.Order{
+				// Create a match with price $49,997.
+				constants.Order_Dave_Num1_Id0_Clob0_Sell1BTC_Price49997_GTB10,
+				constants.Order_Carl_Num1_Id0_Clob0_Buy1BTC_Price50003_GTB10,
+			},
+
+			expectedExistInState: map[clobtypes.OrderId]bool{
+				constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_TP_49995.OrderId: true,
+			},
+			expectedInTriggeredStateAfterBlock: map[uint32]map[clobtypes.OrderId]bool{
+				2: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_TP_49995.OrderId: false},
+				3: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_TP_49995.OrderId: false},
+				4: {constants.ConditionalOrder_Alice_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10_TP_49995.OrderId: false},
+			},
+		},
+		"TakeProfit/Buy conditional order is placed and not triggered by matched price (bounded)": {
```

Comment:
for the first 12 tests (4 untriggered, 4 trigger immediately, 4 trigger in later blocks), added a corresponding test using bounded match prices